pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchConverter.java

```diff
@@ -92,6 +94,7 @@ public static Optional<RawMessage> rebatchMessage(RawMessage msg,
         checkArgument(msg.getMessageIdData().getBatchIndex() == -1);
 
         ByteBuf payload = msg.getHeadersAndPayload();
+        BrokerEntryMetadata brokerEntryMetadata = Commands.parseBrokerEntryMetadataIfExist(payload);
```

Comment:
We can avoid using this method to reduce objects. (read the raw ByteBuf directly).